spdx-utils/src/main/kotlin/model/SpdxPackage.kt

```diff
@@ -0,0 +1,209 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.spdx.model
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.fasterxml.jackson.annotation.JsonProperty
+import org.ossreviewtoolkit.spdx.isSpdxExpressionOrNotPresent
+
+/**
+ * Describes a software package.
+ */
+data class SpdxPackage(
+    /**
+     * A unique identifier for this [SpdxPackage] within a SPDX document.
+     */
+    @JsonProperty("SPDXID")
+    val spdxId: String,
+
+    /**
+     * The [SpdxAnnotation]s for the package.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val annotations: List<SpdxAnnotation> = emptyList(),
+
+    /**
+     * Acknowledgements for the package that may be required to be communicated in some contexts. This is not meant to
+     * include the package's actual complete license text, and may include copyright notices. The SPDX data creator
+     * may use this field to record other acknowledgements, such as particular clauses from license texts, which may be
+     * necessary or desirable to reproduce.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val attributionTexts: List<String> = emptyList(),
+
+    /**
+     * Checksums of the package.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val checksums: List<SpdxChecksum> = emptyList(),
+
+    /**
+     * Any general comments about the package.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val comment: String = "",
+
+    /**
+     * A text relating to a copyright notice, even if not complete. To represent a not present value
+     * [SpdxConstants.NONE] or [SpdxConstants.NOASSERTION] must be used.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val copyrightText: String = "",
+
+    /**
+     * A more detailed description of the package as opposed to [summary], which may be an extract from the package.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val description: String = "",
+
+    /**
+     * The download location as URL. To represent a not present value [SpdxConstants.NONE] or
+     * [SpdxConstants.NOASSERTION] must be used.
+     */
+    val downloadLocation: String,
+
+    /**
+     * References to external sources of additional information.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val externalRefs: List<SpdxExternalReference> = emptyList(),
+
+    /**
+     * Indicates whether the files contents of the package have been used for the creation of the associated
+     * [SpdxDocument].
+     */
+    val filesAnalyzed: Boolean = true,
+
+    /**
+     * The Spdx references to the files belonging to the package in the format "SPDXRef-${id-string}".
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val hasFiles: List<String> = emptyList(),
+
+    /**
+     * The homepage URL. To represent a not present value [SpdxConstants.NONE] or [SpdxConstants.NOASSERTION] must be
+     * used.
```

Comment:
Is this comment actually correct about using `NONE` / `NOASSERTION` here?

Reply:
yes it is, see https://github.com/spdx/spdx-spec/blob/development/v2.2.1/chapters/package-information.md#711-package-home-page-field-.